convert.go

```diff
@@ -100,20 +100,23 @@ func FromIP(ip net.IP) (ma.Multiaddr, error) {
 
 // ToIP converts a Multiaddr to a net.IP when possible
 func ToIP(addr ma.Multiaddr) (net.IP, error) {
-	n, err := ToNetAddr(addr)
+	_, network, ip, _, hostname, err := dialArgComponents(addr)
 	if err != nil {
 		return nil, err
 	}
 
-	switch netAddr := n.(type) {
-	case *net.UDPAddr:
-		return netAddr.IP, nil
-	case *net.TCPAddr:
-		return netAddr.IP, nil
-	case *net.IPAddr:
-		return netAddr.IP, nil
+	if hostname {
+		addr, err := net.ResolveIPAddr(network, ip)
```

Comment:
Hm. I didn't even _realize_ we were resolving these. We probably shouldn't do that. I wouldn't expect `ToIP(...)` to make a network request!

Honestly, I'd just reject this case.

Reply:
This was not happening I think?

Reply:
correct. DialArgs hasn't resolved and returns the string hostname. `ToNetAddr` may do lookups, as it delegates to `ResolveTCPAddr` in the basic codecs, so asking for a NetAddr of a malformed multiaddr like `/tcp4/google.com` would do a DNS resolution for google.

Reply:
Encoding that multiaddr should fail.

Reply:
In latest commit it is changed to fail